server/utils/calendarDay.ts

```diff
@@ -0,0 +1,21 @@
+export default class CalendarDay {
+  constructor(readonly day: number, readonly month: number, readonly year: number) {}
+
+  static parseIso8601(val: string): CalendarDay | null {
+    const result = /^(\d+)-(\d+)-(\d+)$/.exec(val)
```

Comment:
this parses random dash seperated numbers like `1234-8765123-129873`. is that ok? why not specficy length of 4-2-2? (or 2 to 4, 1 to 2, 1 to 2).

`^(\d{4})-(\d{2})-(\d{2})$` (or `^(\d{2-4})-(\d{1-2})-(\d{1-2})$`)

Reply:
in fact, why not use Date.parse(), then we get validation of incorrect dates like 31st february?

const date = new Date(Date.parse(val))
const year = date.getFullYear()
...

Reply:
> is that ok? why not specficy length of 4-2-2

I was being pretty liberal with the parsing - all that I really cared about was being able to split the date into its components so that I can put them into text fields. But I've changed it to use `Date.parse`, not sure whether I prefer it though 😬